### PR TITLE
MAT-365: Move logic for creating donations from controller into new DonationService class

### PR DESCRIPTION
**Pull request by bdsl**
…changing prod code

### CODE REVIEW

File: tests/Application/Actions/Donations/CreateTest.php [hunk 1]
```diff
@@ -906,24 +901,27 @@ public function testCharityLackingCapabilities(): void
...
         // act
         $request = $this->createRequest('POST', TestData\Identity::getTestPersonNewDonationEndpoint(), $data);
         $response = $app->handle($this->addDummyPersonAuth($request));
 
         // assert
-        $payload = (string) $response->getBody();
-        $this->assertJson($payload);
         $this->assertEquals(409, $response->getStatusCode());
 
-        $expectedPayload = new ActionPayload(409, ['error' => [
```
Comment (bdsl) [inline tests/Application/Actions/Donations/CreateTest.php:920]: the 409 here never made it into the serialised string.

File: src/Domain/DonationService.php [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+use MatchBot\Domain\DomainException\CharityAccountLacksNeededCapaiblities;
+use MatchBot\Domain\DomainException\CouldNotMakeStripePaymentIntent;
+use MatchBot\Domain\DomainException\DonationCreateModelLoadFailure;
+use MatchBot\Domain\DomainException\StripeAccountIdNotSetForAccount;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use Random\Randomizer;
+use Stripe\Exception\ApiErrorException;
+use Symfony\Component\Clock\ClockInterface;
+use Symfony\Component\Notifier\Chatter;
```
Comment (NoelLH) [inline src/Domain/DonationService.php:23]:
```suggestion
```

**Review by NoelLH — COMMENTED** [review]
Looks good! Just minor suggestions inline

File: src/Application/Actions/Donations/Create.php [hunk 3]
```diff
@@ -108,272 +80,67 @@ protected function action(Request $request, Response $response, array $args): Re
...
+        } catch (CharityAccountLacksNeededCapaiblities) {
+            return $this->respond(
+                $response,
+                new ActionPayload(
+                // HTTP 409 Conflict can cover when requests conflicts w/ server configuration.
+                    409,
+                    null,
+                    new ActionError(
+                        ActionError::VERIFICATION_ERROR,
+                        'Could not make Stripe Payment Intent (B)'
```
[the remark this answers is not in the document]
Comment (bdsl) [reply src/Application/Actions/Donations/Create.php:121]: Yes - I'll make it C. Done in f83d78a

File: tests/Domain/DonationServiceTest.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Domain;
```
[the remark this answers is not in the document]
Comment (bdsl) [reply tests/Domain/DonationServiceTest.php:3]: Thanks, applied in 424f3a9

File: tests/Domain/DonationServiceTest.php [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+use MatchBot\Application\HttpModels\DonationCreate;
+use MatchBot\Application\Matching\Adapter;
+use MatchBot\Application\Notifier\StripeChatterInterface;
+use MatchBot\Application\Persistence\RetrySafeEntityManager;
+use MatchBot\Client\Stripe;
+use MatchBot\Domain\CampaignRepository;
+use MatchBot\Domain\DomainException\CharityAccountLacksNeededCapaiblities;
+use MatchBot\Domain\DomainException\CouldNotMakeStripePaymentIntent;
```
[the remark this answers is not in the document]
Comment (bdsl) [reply tests/Domain/DonationServiceTest.php:12]: Thanks, removed in 424f3a9

File: src/Domain/DonationService.php (hunk 2 above)
Comment (bdsl) [reply src/Domain/DonationService.php:23]: Thanks, fixed in 86389fa

**Comment by bdsl** [comment]
> Looks good! Just minor suggestions inline

Thanks @NoelLH - all suggestions applied.

**Review by NoelLH — APPROVED** [review]
LGTM!

**Pull request merged**